rpc/src/v1/types/block_number.rs

```diff
@@ -52,6 +52,22 @@ impl BlockNumber {
 			_ => None,
 		}
 	}
+
+	/// Convert block number to block id.
+	pub fn to_block_id(self) -> BlockId {
+		// NOTE Here we treat `Pending` as `Latest`.
+		// Since light clients don't produce pending blocks
+		// (they don't have state) we can safely fallback to `Latest`.
+		match self {
+			BlockNumber::Num(n) => BlockId::Number(n),
+			BlockNumber::Earliest => BlockId::Earliest,
+			BlockNumber::Latest => BlockId::Latest,
+			BlockNumber::Pending => {
+				warn!("`Pending` is deprecated and may be removed in future versions. Falling back to `Latest`");
```

Comment:
Hmm, this should only get logged when accessed from the light client impl. So maybe move back to `num_to_id` on the light client or add a `light_client: bool` method param?

Reply:
Moved it to a trait, so it explicitly says it's for light client only.